src/main/webapp/queue/student.html

```diff
@@ -27,26 +30,52 @@
       var user = firebase.auth().currentUser;
       firebase.auth().onAuthStateChanged(function(user) {
         if (user) {
-          user.getIdToken().then((token) => makeRequest(token));
+          user.getIdToken().then((token) => setToken(token));
         } else {
             console.log("user not logged in");
         }
       });
     }
 
-    function makeRequest(token){
-      var params = window.location.search + "&userToken=" + token;
+    function setToken(token){
+      studentToken = token;
+      makeRequest();
+    }
+
+    function makeRequest(){
+      var params = window.location.search + "&studentToken=" + studentToken;
       const request = new Request("/check-student" + params, {method: "GET"});
-      fetch(request).then(response => response.json()).then((pos) => {
-        document.getElementById("pos").innerHTML = "#" + pos;
+      fetch(request).then(response => response.json()).then((studentPosition) => {
+        if (studentPosition == "0") {
+          document.getElementById("queue_status").innerHTML = "";
+          getTA();
+      
+      fetch(request);
+        } 
+        else {
+            document.getElementById("studentPosition").innerHTML = "#" + studentPosition;
```

Comment:
Its safer to use `.innerText` when what your inserting doesn't have any html in it. This can prevent code injection.